Vizzuality/widget-editor
# Allow the host app to pass a default geographic filter

**⚠️ To be reviewed and merged after #153.** It depends on changes introduced in #153.

This PR allows the host app to pass a default geographic filter to the widget-editor.

This default value will overwrite any set in the widget, but is ignored if the dataset doesn't provide geographic information.

## Testing instructions

1. Edit `src/playground/widget-editor/src/components/editor/component.js` to pass the editor a new props, `areaIntersection`, with the value `'37b397a9a2860beb5c43a1f418d3c3d3'`
2. Restore the dataset `5e156d22-7f84-4cd2-9724-c1a519a83e0a`

Make sure that in the “Geographic filter” section, “Afghanistan” is set by default.

3. Restore the dataset `852f2275-91a8-4500-9f10-89880dc53f22`

Make sure the section “Geographic filter” is not shown, and that in Redux, in the filters, `areaIntersection` is still `null` (this dataset doesn't provide geo info).

4. Restore the dataset `a86d906d-9862-4783-9e30-cdb68cd808b8`

Make sure the filter is also set to “Afghanistan”. This tests that the default filter is applied when the dataset is changed.

5. Restore the widget `df070ec2-b913-4d80-ae51-923694a9a250` (dataset: `5e156d22-7f84-4cd2-9724-c1a519a83e0a`)

Make sure the filter is also set to “Afghanistan”. This tests that even if a widget already has a geo filter, it is overwritten.

6. In `src/playground/widget-editor/src/components/editor/component.js`, change the `areaIntersection` prop to `'01a4460b5ae0a6688b27a0ee987394dc'`
7. Restore the dataset `5e156d22-7f84-4cd2-9724-c1a519a83e0a`

Make sure that the “Geographic filter” section is present, and that it is set to “Custom area”. This tests that custom areas can be used as default filters.

## Pivotal Tracker

[Task](https://www.pivotaltracker.com/story/show/174968416).
